docs/testplan/dhcp_relay/DHCPv4-Relay-Test-Plan.md

```diff
@@ -0,0 +1,338 @@
+      - [2.2 Uplinks Down at Start Test (`test_dhcp_relay_start_with_uplinks_down`)](#22-uplinks-down-at-start-test-test_dhcp_relay_start_with_uplinks_down)
+    - [3. Advanced Features Tests](#3-advanced-features-tests)
+      - [3.1 Unicast MAC Test (`test_dhcp_relay_unicast_mac`)](#31-unicast-mac-test-test_dhcp_relay_unicast_mac)
+      - [3.2 Random Source Port Test (`test_dhcp_relay_random_sport`)](#32-random-source-port-test-test_dhcp_relay_random_sport)
+      - [3.3 Counter Validation Test (`test_dhcp_relay_counter`)](#33-counter-validation-test-test_dhcp_relay_counter)
+      - [3.4 max-hop-count Test (`test_dhcp_max_hop_count`)](#34-max-hop-count-test-test_dhcp_max_hop_count)
+    - [4. Option 82 and Agent Mode Tests](#4-option-82-and-agent-mode-tests)
+      - [4.1 Option 82 Sub-options Test (`test_dhcp_relay_option82_suboptions`)](#41-option-82-sub-options-test-test_dhcp_relay_option82_suboptions)
+      - [4.2 Agent Mode Test (`test_dhcp_relay_agent_mode`)](#42-agent-mode-test-test_dhcp_relay_agent_mode)
+    - [5. VRF Support Tests](#5-vrf-support-tests)
+      - [5.1 Non-Default VRF Test (`test_dhcp_relay_with_non_default_vrf`)](#51-non-default-vrf-test-test_dhcp_relay_with_non_default_vrf)
+      - [5.2 Different VRF Test (`test_dhcp_relay_with_different_non_default_vrf`)](#52-different-vrf-test-test_dhcp_relay_with_different_non_default_vrf)
+  - [Test Utilities and Fixtures](#test-utilities-and-fixtures)
+    - [Key Helper Functions](#key-helper-functions)
+    - [Fixtures](#fixtures)
+  - [Validation Criteria](#validation-criteria)
+    - [DHCP Message Flow](#dhcp-message-flow)
+    - [Counter Validation](#counter-validation)
+    - [Log Analysis](#log-analysis)
+    - [Performance Requirements](#performance-requirements)
+  - [Known Limitations](#known-limitations)
+    - [Version Restrictions](#version-restrictions)
+    - [Feature Restrictions](#feature-restrictions)
+  - [Troubleshooting Guide](#troubleshooting-guide)
+    - [Common Issues](#common-issues)
+    - [Debug Commands](#debug-commands)
+
+## Document Information
+- **Test Suite**: SONiC DHCPv4 relay Testing
+- **Files**: `tests/dhcp_relay/test_dhcp_relay.py`, `tests/dhcp_relay/test_dhcpv4_relay.py`
+- **Date**: June 25, 2025
+- **Version**: 1.0
+
+## References
+
+| **Document Name** | **Link** |
+|-------------------|----------|
+| SONIC DHCPv4 Relay | [DHCPv4_relay/DHCPv4-relay-agent-High-Level-Design.md](https://github.com/sonic-net/SONiC/blob/master/doc/dhcp_server/port_based_dhcp_server_high_level_design.md)|
+
+## Overview
+
+This test plan covers comprehensive testing of DHCPv4 relay functionality in SONiC switches. The test suite validates both ISC DHCPv4 relay agent and SONiC DHCPv4 relay agent implementations across various network topologies and configurations.
+
+## Test Environment
+
+### Topology Requirements
+
+  Supported Topologies: T0, dual-tor
```

Comment:
We have dhcp_relay enabled on m0 topo as well. The topo is pretty similar to T0.

Reply:
@Blueve Do we run sonic-mgmt tests separately on m0 topology?

Reply:
@Blueve I've added M0 to the list of supported topologies. We expect the functionality to work irrespective of the topology.